source/MaterialXFormat/Export.h

```diff
@@ -0,0 +1,22 @@
+//
+// TM & (c) 2021 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#ifndef MATERIALX_FORMAT_EXPORT_H
+#define MATERIALX_FORMAT_EXPORT_H
+
+#include <MaterialXCore/Library.h>
+
+/// @file
+/// Macros for declaring imported and exported symbols.
+
+#if defined(MATERIALX_FORMAT_EXPORTS)
+    #define MX_FORMAT_API MATERIALX_SYMBOL_EXPORT
```

Comment:
A quick question here. Is there a need to have separate macros for each library. For example, can there just be an `MX_API` instead of `MATERIALX_SYMBOL_EXPORT` and not have `MX_CORE_API`, `MX_FORMAT_API` etc.

Just wondering if there a case where only specific libraries will have their symbols exported but not others ?

Reply:
The separate macros per-library are not too allow only specific libraries to export their symbols. It's because the different MaterialX libraries import symbols from each other. So Format imports Core symbols while exporting Format symbols. So when building the Format library, MX_FORMAT_API is set to "export", but MX_CORE_API is set to "import".

Reply:
Yes. Jonathan explained this context information as well -- which I didn't see. Please disregard this comment  :)